lib/constructs/interface.js

```diff
@@ -130,6 +132,52 @@ Interface.prototype.generateExport = function () {
     exposers.push(keys[i] + ": { " + exposedOnObj.join(", ") + " }");
   }
 
+  if (this.iterable) {
+    this.str += `\nconst IteratorPrototype = Object.create(utils.IteratorPrototype, {
+  next: {
+    value: function next() {
+      const internal = this[utils.iterInternalSymbol];
+      const target = internal.target;
+      const kind = internal.kind;
+      const index = internal.index;
+      const values = Array.from(target[impl]);
```

Comment:
This is really just a bare-bones implementation of the spec. Not very efficient for anything that is not stored as an array in the impl, but the spec does ask for a "list of value pairs to iterate over", which in ECMAScript should be expressed as an array.

One thing that could be optimized is the `switch` statement involving `kind`. We could create three functions for each `kind` value and then store the callback pointer in the internal object